### PR TITLE
Replace 500 errors with more meaningful error codes

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
Replace 500 errors with more meaningful codes. What seemed to be the root cause of the errors that led to the creation of the ticket was fixed by https://github.com/mattermost/customer-web-server/pull/899 , however this PR replaces these generic 500 errors with more meaningful error codes.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46873

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: api4/cloud.go [hunk 1]
```diff
@@ -244,14 +244,14 @@ func validateWorkspaceBusinessEmail(c *Context, w http.ResponseWriter, r *http.R
 
 	user, userErr := c.App.GetUser(c.AppContext.Session().UserId)
 	if userErr != nil {
-		c.Err = model.NewAppError("Api4.validateWorkspaceBusinessEmail", "api.cloud.request_error", nil, userErr.Error(), http.StatusInternalServerError)
+		c.Err = model.NewAppError("Api4.validateWorkspaceBusinessEmail", "api.cloud.request_error", nil, userErr.Error(), http.StatusForbidden)
```
[the remark this answers is not in the document]
Comment (pvev) [reply api4/cloud.go:247]: ok! nice catch, thanks.

File: model/client4.go [hunk 2]
```diff
@@ -8007,7 +8007,7 @@ func (c *Client4) ConfirmCustomerPayment(confirmRequest *ConfirmPaymentMethodReq
 func (c *Client4) RequestCloudTrial(email *StartCloudTrialRequest) (*Subscription, *Response, error) {
 	payload, err := json.Marshal(email)
 	if err != nil {
-		return nil, nil, NewAppError("RequestCloudTrial", "api.marshal_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		return nil, nil, NewAppError("RequestCloudTrial", "api.marshal_error", nil, "", http.StatusBadRequest).Wrap(err)
```
[the remark this answers is not in the document]
Comment (pvev) [reply model/client4.go:8010]: ok, you are right, it doesn't make that much sense to replace this for bad_request for the json.marshal error. I will keep it as InternalServerError and continue being standard with all the other error handling existing in the web-server code.

**Comment by pvev** [comment]
@marianunez :)

**Review by marianunez — APPROVED** [review]

**Pull request merged**